internal/commands/buildpack_new.go

```diff
@@ -66,11 +71,46 @@ func BuildpackNew(logger logging.Logger, creator BuildpackCreator) *cobra.Comman
 				})
 			}
 
+			var targets []dist.Target
+			for _, t := range flags.Targets {
+				var distroMap []dist.Distribution
+				target, nonDistro, distros, err := getTarget(t)
+				if err != nil {
+					logger.Error(err.Error())
+				}
+				if i, e := getSliceAt[string](target, 1); e == nil {
+					distros = strings.Split(i, ";")
+				}
+				for _, d := range distros {
+					distro := strings.Split(d, "@")
+					if l := len(distro); l <= 0 {
+						return errors.Errorf("distro is nil!")
+					} else if l == 1 {
+						logger.Warnf("forgot to specify version for distro %s ?", distro[0])
+					}
+					distroMap = append(distroMap, dist.Distribution{
+						Name:     distro[0],
+						Versions: distro[1:],
+					})
+				}
+				os, arch, variant, err := getPlatform(nonDistro)
+				if err != nil {
+					return err
+				}
+				targets = append(targets, dist.Target{
+					OS:            os,
+					Arch:          arch,
+					ArchVariant:   variant,
+					Distributions: distroMap,
+				})
+			}
```

Comment:
I will suggest a quick refactor, the reason behind that is because your code to parse the platform input could be reuse in other commands on pack where are working to use the pattern. 😄 

I will suggest to create a `internal/target` package with two files `parse.go` and `platform.go`. 

In `parse.go` move

```golang 
func ParseTargets(targetsInput []string, logger logging.Logger) ([]dist.Target, error) {
   // move all this logic from L74 to L106
}

func getTarget(t string) ([]string, []string, []string, error) { }

func getSliceAt[T interface{}](slice []T, index int) (T, error) { }

```

In `platform.go` move
 
```golang
var GOOSArch = map[string][]string{ }             // I think this can be internal to the package
var GOArchVariant = map[string][]string{ }      // also, can be rename to lower case to be internal to the package

func getPlatform(t []string) (string, string, string, error) { }

func isOS(os string) bool { }

func supportsArch(os string, arch string) bool { }

func supportsVariant(arch string, variant string) bool { }

```


Then you can replace with something like:

```golang
	targets, err := target.ParseTargets(flags.Targets, logger)
	if err != nil {
		return err
	}
```

If you want, you can also simplify your test cases, because now you can focus on your function `ParseTargets`. I want to suggest the function to return the parsed strings, but then I noticed we have the `dist.Distributions` and make things complicated, so it is fine to return the `dist.Target` object.

Reply:
done :)